presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -125,7 +127,22 @@ public double getOutputSizeInBytes()
      * Returns estimated data size.
      * Unknown value is represented by {@link Double#NaN}
      */
-    public double getOutputSizeInBytes(Collection<VariableReferenceExpression> outputVariables)
+    public double getOutputSizeInBytes(PlanNode planNode)
+    {
+        requireNonNull(planNode, "planNode is null");
+
+        if (sourceInfo instanceof HistoryBasedSourceInfo && !isNaN(totalSize) && confident) {
```

Comment:
not of big fan of downcast; shouldn't we have a method for `SourceInfo`? Also, I guess `SourceInfo` is an interface instead of an abstract class?

Reply:
+1